### ISSUE TITLE
Expr: log(0) returns -87.3365, instead of -nan

### ISSUE THREAD

**Issue by AkarinVS** [vapoursynth/vapoursynth#705]
The current Expr implementation computes `0 log` as -87.3365, but log of any negative number is `-nan`.
This is inconsistent. Should we fix it? Or is it intentional?

```python
import vapoursynth as vs
core = vs.core
c = core.std.BlankClip(format=vs.RGBS, color=[-1,0,1])
c = core.std.Expr(c, 'x log')
print([c.get_frame(0).get_read_array(i)[0,0] for i in range(3)])
# it prints [nan, -87.3365478515625, 0.0]
```

Thanks.

**Comment by myrsloik** [comment]
I don't think it matters since the value effectively is undefined and any expression relying on it is questionable at best.

**Comment by AkarinVS** [comment]
In reality it probably doesn't matter, but I think the current behavior is
an oversight in the code.
Should we fix it? It's a matter of flipping a comparison operator.
As you said, no code should rely on this behavior, so fixing it shouldn't
break existing code.

I'd like to include this fix and remove another case of obvious dead code
in my PR #693
(as separate commits so that git blame will work as expected, of course).
What do you think?

**Comment by myrsloik** [comment]
If the change is that small, sure. go for it.

**Comment by sekrit-twc** [comment]
It's not intentional, but can this be solved without burdening the generated code?

**Comment by AkarinVS** [comment]
On Sun, Jun 27, 2021 at 12:19 PM sekrit-twc ***@***.***>
wrote:

> It's not intentional, but can this be solved without burdening the
> generated code?
>
We just need to change a comparison from "x < zero" to "x <= zero", so
fixing the issue won't burden the generated code.

**Comment by sekrit-twc** [comment]
OK, let's fix it.

**Comment by AkarinVS** [comment]
Fix is included in PR #693.

>
